presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -575,18 +605,27 @@ private PlanBuilder aggregate(PlanBuilder subPlan, QuerySpecification node)
                             entry.getValue(),
                             builder.build(),
                             Optional.empty(),
-                            ImmutableList.of()));
+                            aggregations.values().stream()
+                                    .map(Aggregation::getOrderBy)
+                                    .flatMap(Collection::stream)
+                                    .collect(toImmutableList())));
         }
 
         AggregationNode aggregationNode = new AggregationNode(
                 idAllocator.getNextId(),
                 subPlan.getRoot(),
-                aggregations.build(),
+                aggregations,
                 groupingSymbols,
                 AggregationNode.Step.SINGLE,
                 Optional.empty(),
                 groupIdSymbol);
 
+        if (aggregationNode.hasEmptyGroupingSet() && aggregationNode.hasNonEmptyGroupingSet() && !aggregationNode.getOrderBySymbols().isEmpty()) {
+            throw SemanticExceptions.notSupportedException(
+                    node,
+                    "ORDER BY in aggregate function with empty grouping set");
```

Comment:
Why should it require partial aggregation? Isn't empty grouping set equivalent to grouping over constant? This should work as long as data size fits into memory, but ... this is true for any kind of aggregation.